app/controllers/concerns/idv/threat_metrix_concern.rb

```diff
@@ -0,0 +1,59 @@
+# frozen_string_literal: true
+
+module Idv
+  module ThreatMetrixConcern
+    THREAT_METRIX_DOMAIN = 'h.online-metrix.net'
+    THREAT_METRIX_WILDCARD_DOMAIN = '*.online-metrix.net'
+
+    def override_csp_for_threat_metrix
+      return unless Rails.env.production?
+
+      return unless IdentityConfig.store.proofing_device_profiling_collecting_enabled
+
+      return if params[:step] != 'ssn'
+
+      policy = current_content_security_policy
+
+      # ThreatMetrix requires additional Content Security Policy (CSP)
+      # directives to be added to the response to enable its JS to run
+      # in the browser.
+
+      # `script-src` must be updated to enable:
+      #   - The domain hosting ThreatMetrix JS (so it can be included on the page)
+      #   - `unsafe-eval`, since the ThreatMetrix JS uses eval() internally.
+      add_to_policy policy, :script_src, THREAT_METRIX_DOMAIN, :unsafe_eval
+
+      # `style-src` must be update to enable:
+      #   - `unsafe-inline`, since the ThreatMetrix library applies inline
+      #      styles to elements it inserts into the DOM
+      add_to_policy policy, :style_src, :unsafe_inline
+
+      # `img-src` must be updated to enable:
+      #   - The domain hosting ThreatMetrix JS, since ThreatMetrix loads
+      #     images from this domain as part of its fingerprinting process.
+      add_to_policy policy, :img_src, THREAT_METRIX_WILDCARD_DOMAIN
+
+      # `connect-src` must be updated to enable:
+      #   - The domain hosting ThreatMetrix JS, since ThreatMetrix makes XHR
+      #     requests to this domain.
+      add_to_policy policy, :connect_src, THREAT_METRIX_DOMAIN
+
+      # `child-src` must be updated to enable:
+      #   - The domain hosting ThreatMetrix JS, which used to load a fallback
+      #     `<iframe>` element when Javascript is disabled.
+      add_to_policy policy, :child_src, THREAT_METRIX_DOMAIN
+
+      request.content_security_policy = policy
+    end
+
+    private
+
+    def add_to_policy(policy, directive, *values)
+      existing_values = policy.directives[directive]
+
+      new_values = [existing_values, values].flatten.uniq.compact
+
+      policy.send(directive, *new_values)
```

Comment:
I don't super love `#send` if we can avoid it -- is that because we wanted to DRY this "add to existing" bit?

would something like this work for each, to avoid send but keep things relatively terse?

```ruby
policy.child_src(*(policy.child_src.to_set << THREAT_METRIX_DOMAIN))
```

Reply:
It gets a little gunky because `policy.child_src` [is not a getter](https://github.com/rails/rails/blob/main/actionpack/lib/action_dispatch/http/content_security_policy.rb#L179-L185), so you have to read `policy.directives['child-src']`, giving you something like:

```ruby
policy.child_src(*(policy.directives['child-src'].to_set << THREAT_MATRIX_DOMAIN))
```

which for me (admittedly not a rubyist) triggers a lot of "there are a lot of weird characters in that line" alarm

Reply:
reading that code, it looks like `policy.child_src` is a getter because it's inside the [DIRECTIVES](https://github.com/rails/rails/blob/08af60e01dcf33db34a5ecafab8be25e373f7bfb/actionpack/lib/action_dispatch/http/content_security_policy.rb#L140) constant? 

We do something similar here for all the other directives (and don't even worry about dupes, which would make the code even more terse)

https://github.com/18F/identity-idp/blob/38a8474297f0d2de9fd583ae123396af8d3a91dc/app/controllers/concerns/idv/document_capture_concern.rb#L7-L10